docs/USING_PUBLIC_CLOUDS.rst

```diff
@@ -0,0 +1,255 @@
+
+.. code:: shell
+
+   # Start the server locally
+   $> S3DATA=multiple npm start
+
+Run the server as a docker container with the ability to write to AWS S3
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Mount all the files that have been edited to override defaults, and do a
+standard Docker run; then you can start testing pushing to AWS S3.
+
+.. code:: shell
+
+   # Start the server in a Docker container
+   $> sudo docker run -d --name CloudServer \
+   -v $(pwd)/data:/usr/src/app/localData \
+   -v $(pwd)/metadata:/usr/src/app/localMetadata \
+   -v $(pwd)/locationConfig.json:/usr/src/app/locationConfig.json \
+   -v $(pwd)/conf/authdata.json:/usr/src/app/conf/authdata.json \
+   -v ~/.aws/credentials:/root/.aws/credentials \
+   -e S3DATA=multiple -e ENDPOINT=http://localhost -p 8000:8000
+   -d scality/s3server
+
+Testing: put an object to AWS S3 using CloudServer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to start testing pushing to AWS S3, you will need to create a local
+bucket in the AWS S3 region - this local bucket will only store the metadata
+locally, while both the data and the metadata will be stored on AWS S3.
+This example is based on all our previous steps.
+
+.. code:: shell
+
+   # Create a local bucket hosting data in AWS S3
+   $> s3cmd --host=127.0.0.1:8000 mb s3://zenkobucket --region=aws-test
+   # Put an object to AWS S3, and store the metadata locally
+   $> s3cmd --host=127.0.0.1:8000 put /etc/hosts s3://zenkobucket/testput
+    upload: '/etc/hosts' -> 's3://zenkobucket/testput'  [1 of 1]
+     330 of 330   100% in    0s   380.87 B/s  done
+   # List locally to check you have the metadata
+   $> s3cmd --host=127.0.0.1:8000 ls s3://zenkobucket
+    2017-10-23 10:26       330   s3://zenkobucket/testput
+
+Then, from the AWS Console, if you go into your bucket, you should see your
+newly uploaded object:
+
+.. figure:: ../res/aws-console-successful-put.png
+   :alt: AWS S3 Console upload example
+
+Troubleshooting
+~~~~~~~~~~~~~~~
+
+Make sure your :code:`~/.s3cfg` file has credentials matching your local
+CloudServer credentials defined in :code:`conf/authdata.json`. By default, the
+access key is :code:`accessKey1` and the secret key is :code:`verySecretKey1`
+
+Make sure versioning is enabled in your remote AWS S3 hosted bucket. To check,
+using the AWS Console, click on your bucket name, then on "Properties" at the
+top, and then you should see something like this:
+
+.. figure:: ../res/aws-console-versioning-enabled.png
+   :alt: AWS Console showing versioning enabled
+
+Microsoft Azure as a backend
+----------------------------
+
+locationConfig.json
+~~~~~~~~~~~~~~~~~~~
+
+Add a new region, which should describe the MS Azure hosted container you'll be
+writing your data to. There are a few configurable options here:
+
+- :code:`type` : set to :code:`azure` to indicate this region is hosted in MS Azure;
+- :code:`legacyAwsBehavior` : set to :code:`false`;
```

Comment:
'legacyAwsBehavior' might need to be 'true', @dora-korpar ?

Reply:
Yeah, we have all Azure backends set to 'true' for consistency during testing. It doesn't really matter either way what it's set to; it's more for our S3 to return errors the way the 'us-east-1' AWS region does. Just depends on what functionality the user wants

Reply:
OK, interesting. I've done all my tests with legacyAwsBehavior set to false, hence why I set it to that there. I'll just add the description I put for AWS S3 then, sounds good?